rig-neo4j/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+use neo4rs::*;
+use std::fmt::Debug;
+
+use rig::{
+    embeddings::{DocumentEmbeddings, Embedding, EmbeddingModel},
+    vector_store::{VectorStore, VectorStoreError, VectorStoreIndex},
+};
+use serde::Deserialize;
+
+pub struct Neo4jClient {
+    pub graph: Graph,
+}
+
+/// A Neo4j vector store.
+pub struct Neo4jVectorStore {
+    //collection: mongodb::Collection<DocumentEmbeddings>,
+    pub client: Neo4jClient,
+    pub database_name: String,
+}
+
+fn neo4j_to_rig_error(e: neo4rs::Error) -> VectorStoreError {
+    VectorStoreError::DatastoreError(Box::new(e))
+}
```

Comment:
I think you should be able to use `thiserror` to avoid manually `Box`'ing errors when converting from library errors to crate errors